### PR TITLE
[Fixes #77] Push improvements to configurations

### LINKED ISSUE #77: Push improvements to configurations
@afabiani please create a PR (here and toward their upgrade_3.3.x branch) with the list of things we noted during the call:

 - fix the requirements to the latest stable GN 3.3.x
 - make it work async and improve the celery cmd parameters

I would leave the DEBUG=True for the moment, They will change it themselves for their prod deployment.

### PR DESCRIPTION
**Pull request by afabiani**

### CODE REVIEW

File: docker/letsencrypt/crontab [hunk 1]
```diff
@@ -1,9 +1,8 @@
-# .............. minute (0 - 59)
-# .     .............. hour (0 - 23)
-# .     .     .............. day of month (1 - 31)
-# .     .     .     .............. month (1 - 12)
-# .     .     .     .     .............. day of week (0 - 6) (Sunday to Saturday; 7 is also Sunday on some systems)
-# .     .     .     .     .
+# +------------- minute (0 - 59)
+# �     +------------- hour (0 - 23)
+# �     �     +------------- day of month (1 - 31)
```
Comment (mattiagiupponi) [inline docker/letsencrypt/crontab:3]: Is the � an error due to the GitHub view or is it a real unwanted character in the code?

File: src/celery.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 nohup celery -A geonode.celery_app:app beat -l DEBUG -f /var/log/celery.log &>/dev/null &
-nohup celery -A geonode.celery_app:app worker --without-gossip --without-mingle -Ofair -B -E -s django_celery_beat.schedulers:DatabaseScheduler --loglevel=INFO --concurrency=2 -n worker1@%h -f /var/log/celery.log &>/dev/null &
+nohup celery -A geonode.celery_app:app worker --without-gossip --without-mingle -Ofair -B -E --statedb=worker.state --scheduler=celery.beat:PersistentScheduler --loglevel=INFO --concurrency=2 -n worker1@%h -f /var/log/celery.log &>/dev/null &
```
Comment (mattiagiupponi) [inline src/celery.sh:3]: Here we are not considering the `CELERY_WORKER_CONCURRENCY `env var, should we replace it even here?

File: src/tasks.py [hunk 3]
```diff
@@ -176,20 +176,21 @@ def update(ctx):
 def migrations(ctx):
     print("**************************migrations*******************************")
     ctx.run(f"python manage.py migrate --noinput --settings={_localsettings()}", pty=True)
-    print("**************************datastore*******************************")
-    ctx.run(f"python manage.py migrate --noinput --database datastore --settings={_localsettings()}", pty=True)
```
Comment (mattiagiupponi) [inline src/tasks.py:180]: this was needed for the dynamic model schema creation. The migration must be apply on both databases

**Review by mattiagiupponi — CHANGES_REQUESTED** [review]

File: src/celery.sh (hunk 2 above)
Comment (afabiani) [reply src/celery.sh:3]: Not sure, again I just did a copy from the current geonode-project 3.3.x template so maybe we should first make this change upstream?

File: src/tasks.py (hunk 3 above)
Comment (afabiani) [reply src/tasks.py:180]: I see, my fault, I'll revert this one, thanks

Comment (afabiani) [reply src/tasks.py]: Reverted

File: src/celery.sh (hunk 2 above)
Comment (afabiani) [reply src/celery.sh:3]: @mattiagiupponi that file is not the one used by docker, it is just a sample .sh file. `celery-cmd` takes the concurrency variable into account already.

**Review by mattiagiupponi — CHANGES_REQUESTED** [review]
@afabiani there are some conflicts in the settings

**Review by mattiagiupponi — APPROVED** [review]

**Pull request merged**